app/services/smart_annotations/permission_eval.rb

```diff
@@ -26,7 +26,8 @@ def validate_tsk_permissions(user, team, object)
 
       def validate_rep_item_permissions(user, team, object)
         if object.repository
-          return object.repository.team.id == team.id &&
+          return (object.repository.team.id == team.id ||
+                 object.repository.team_repositories.where(team_id: team.id).first.try(:team_id) == team.id) &&
```

Comment:
This check is the same as the one in `can_read_repository?`:
```
repository.team_repositories.where(team: user.teams).any?
```

You filter with `where(team_id: team.id)` so equality check is unneeded, only presence.